### ISSUE TITLE
`jax.numpy.tanh` default JVP loses finite float64 tail derivatives that JAX's highest-accuracy rule preserves

### ISSUE THREAD

**Issue by beanduan22** [jax-ml/jax#40343]
### Description

`jax.grad(jax.numpy.tanh)` returns zero at `x = ±20.0`, although the analytic derivative is a finite normal float64 value:

```text
d/dx tanh(x) = 4 * exp(-2|x|) / (1 + exp(-2|x|))**2
             = 1.6993417021166355e-17 at |x| = 20
```

JAX's `lax.tanh` with `accuracy=lax.AccuracyMode.HIGHEST` returns the expected nonzero derivative at `x = 20.0`, while the default public API returns zero.

```python
import math
import jax
import jax.numpy as jnp
from jax import lax

jax.config.update("jax_enable_x64", True)
default_grad = jax.grad(jnp.tanh)
highest_grad = jax.grad(lambda x: lax.tanh(x, accuracy=lax.AccuracyMode.HIGHEST))
for x_value in (20.0, -20.0):
    x = jnp.asarray(x_value, dtype=jnp.float64)
    print("x:", x_value, "default:", float(default_grad(x)))
print("highest-accuracy at +20:", float(highest_grad(jnp.float64(20.0))))
print("expected:", 4.0 * math.exp(-40.0) / (1.0 + math.exp(-40.0)) ** 2)
```

## Actual result

On JAX `0.9.2`, CPU and two CUDA devices reproduce:

```text
x: 20.0 default: 0.0
x: -20.0 default: 0.0
highest-accuracy at +20: 1.6993417021166355e-17
expected: 1.6993417021166355e-17
```

## Expected result

The default `jax.numpy.tanh` derivative should remain finite and close to `1.6993417021166355e-17` while that value is representable, or provide a public accuracy control equivalent to `lax.tanh(..., accuracy=HIGHEST)`.

### System info (python version, jaxlib version, accelerator, etc.)

```text
JAX version: 0.9.2
jaxlib version: 0.9.2
Python version: 3.13.5
Input dtype: float64, with jax_enable_x64=True
Devices tested: CPU and two CUDA GPUs
```

**Comment by jakevdp** [comment]
Thanks for the report – this looks to fall under #38651.

**Comment by hawkinsp** [comment]
~I think the documentation in https://github.com/jax-ml/jax/pull/40414 fixed this~

We just need to clarify the documentation for the float64 case.
